Sources/AppleMapsKit/Authorization/TokenResponse.swift

```diff
@@ -0,0 +1,24 @@
+import Foundation
+
+/// An object that contains an access token and an expiration time in seconds.
+struct TokenResponse: Codable {
+    /// A string that represents the access token.
+    let accessToken: String
+
+    /// An integer that indicates the time, in seconds from now until the token expires.
+    let expiresInSeconds: Int
+}
+
+extension TokenResponse {
+    
+    /// A boolean indicates weather to token is valid 10 seconds before it's actual expiry time.
+    var isValid: Bool {
```

Comment:
I think this doesn't ever become false, as the `Date.now` is new every time and `expiresInSeconds` doesn't get update since the response is received.

Maybe we can add a private `expirationDate` property to `TokenResponse` to which we assign `Date.now.addingTimeInterval(TimeInterval(expiresInSeconds))` when the `TokenResponse` gets initialized

Reply:
Yes it's a very good point, sorry for that.
I will push a fix and write a test for it.